docs/presentations/linkedin-meetup-2024-04-24.qmd

```diff
@@ -0,0 +1,549 @@
+```
+
+A serverless, highly scalable, and cost-effective cloud data warehouse.
+
+## Snowflake {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.snowflake.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A cloud data platform.
+
+## Oracle {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.oracle.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A relational database management system.
+
+## Spark {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.pyspark.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A unified analytics engine for large-scale data processing.
+
+## Trino {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.trino.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A distributed SQL query engine.
+
+## Flink {auto-animate="true"}
+
+```python
+import ibis
+
+con = ibis.flink.connect()
+penguins = con.table("penguins")
+penguins.group_by("species", "island").agg(count=ibis._.count())
+```
+
+A distributed streaming and batch SQL analytics engine.
+:::
+
+# How does Ibis work?
+
+This is going to be very fast 🏃💨. I will happily answer questions about it 😂.
+
+## Components
+
+- Expression system
+- Compiler
+- Driver
+
+## Expressions
+
+- Expr vs op
+- Type system
+- Pattern matching
+- Traversal
+
+## Expressions cont.
+
+```{python}
+#| echo: true
+expr = (
+  pengs.group_by("species")
+  .mutate(s.across(s.numeric() & ~s.c("year"), (_ - _.mean()) / _.std()))
+  .head(5)
+)
+```
+
+Graph representation
+
+```{python}
+from ibis.expr.visualize import to_graph
+
+to_graph(expr)
+```
+
+## Compilers
+
+- Big refactor in 9.0
+- Broadly speaking, toposort the expression graph
+- Compile a node -> store the output -> pass that to the next node as input -> repeat until the list of nodes is exhausted
+- Show a code example
+
+# Streaming
+
+## Slide 1
+## Slide 2
+## Slide 3
+## Slide …
+## Slide N
+
+## Try it out now
+
+Install:
+
+```bash
+pip install 'ibis-framework[duckdb]'
+```
+
+Then run:
+
+```{python}
+#| echo: true
+import ibis
+
+ibis.options.interactive = True
+
+t = ibis.examples.penguins.fetch()
+
+t
+```
+
+## Questions?
+
+### Where to find us
+
+:::: {.columns}
+::: {.column width="50%"}
+Phillip
+
+- [LinkedIn (phillip-cloud)](https://www.linkedin.com/in/phillip-cloud/)
+- [GitHub (@cpcloud)](https://github.com/cpcloud)
+- [X (@cpcloudy)](https://twitter.com/cpcloudy)
+
+:::
+::: {.column width="50%"}
+Chloe
+
+- A
+- B
+- C
+
+:::
+::::
+
+### Links
+
+- GitHub: <https://github.com/ibis-project/ibis>
+- Site: <https://ibis-project.org>
+- Installation: <https://ibis-project.org/install>
+- Tutorial: <https://ibis-project.org/tutorials/getting_started>
+
```

Comment:
also probably want to add a link to this presentation (if the URL is finalized)

Reply:
Done.